airbyte-workers/src/main/java/io/airbyte/workers/internal/SingleStateAggregator.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.internal;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.State;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import java.util.List;
+
+public class SingleStateAggregator implements StateAggregator {
+
+  AirbyteStateMessage state;
+
+  @Override
+  public void ingest(final AirbyteStateMessage stateMessage) {
+    final AirbyteStateType stateType = stateMessage.getType();
+    Preconditions.checkArgument(stateType == AirbyteStateType.GLOBAL || stateType == AirbyteStateType.LEGACY || stateType == null);
```

Comment:
I think those one are useless. I will change change the scope of the class and move it to its own package.